.github/workflows/ci.yaml

```
@@ -75,8 +139,17 @@ jobs:
           juju-channel: 3.6/stable
           charmcraft-channel: 3.x/stable
 
-      - run: |
-          sg snap_microk8s -c "tox -e ${{ matrix.charm }}-integration -- -- --model testing"
+      - name: Download packed charm(s)
+        id: download-charms
+        timeout-minutes: 5
+        uses: actions/download-artifact@v4
+        with:
+          pattern: ${{ needs.build.outputs.artifact-prefix }}-*
+          merge-multiple: true
+
+      - name: Integration tests
+        run: |
+          sg snap_microk8s -c "tox -e ${{ matrix.charm }}-integration -- --model testing --charm-path=${{ github.workspace }}/charms/${{ matrix.charm }}/${{ matrix.charm }}_ubuntu@20.04-amd64.charm"
```

nit: increase verbosity
```suggestion
          sg snap_microk8s -c "tox -vve ${{ matrix.charm }}-integration -- --model testing --charm-path=${{ github.workspace }}/charms/${{ matrix.charm }}/${{ matrix.charm }}_ubuntu@20.04-amd64.charm"
```